### PR TITLE
[storage] add counters for latest committed/snapshot version

### PR DESCRIPTION
**Pull request by lightmark**
### Description
To consolidate our dashboard.

### Test Plan
ut

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2368)

### CODE REVIEW

**Review by grao1991 — APPROVED** [review]

File: storage/aptosdb/src/metrics.rs [hunk 1]
```diff
@@ -137,6 +137,23 @@ pub static ROCKSDB_PROPERTIES: Lazy<IntGaugeVec> = Lazy::new(|| {
     .unwrap()
 });
 
+// Async committer gauges:
+pub(crate) static LATEST_SNAPSHOT_VERSION: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
+        "aptos_latest_snapshot_version",
```
Comment (msmouse) [inline storage/aptosdb/src/metrics.rs:143]: "state_snapshot"

also I think it's good to keep aptos_storage_ prefix to match other metrics in this crate

File: storage/aptosdb/src/metrics.rs [hunk 2]
```diff
@@ -137,6 +137,23 @@ pub static ROCKSDB_PROPERTIES: Lazy<IntGaugeVec> = Lazy::new(|| {
...
+        "The version of the most recent snapshot."
+    )
+    .unwrap()
+});
+
+pub(crate) static LATEST_CHECKPOINT_VERSION: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
+        "aptos_latest_checkpoint_version",
```
Comment (msmouse) [inline storage/aptosdb/src/metrics.rs:151]: "state_checkpoint"

File: storage/aptosdb/src/state_store/buffered_state.rs [hunk 3]
```diff
@@ -116,6 +120,12 @@ impl BufferedState {
         self.maybe_commit(true /* sync_commit */);
     }
 
+    fn report_latest_committed_version(&self) {
+        if let Some(version) = self.state_after_checkpoint.base_version {
```
Comment (msmouse) [inline storage/aptosdb/src/state_store/buffered_state.rs:124]: When is this `None`? Only when DB is empty? I'd write a -1 for that.

**Review by msmouse — APPROVED** [review]
beautiful

**Comment by gregnazario** [comment]
@lightmark Please rebase

**Comment by lightmark** [comment]
@gregnazario I thought this had landed...

**Pull request merged**